### DISCUSSION TITLE
Flows - possible to add event to trigger object?

### DISCUSSION THREAD

**Discussion by ddelyon** [directus/directus#14048] [Archive]
Loving the Flows functionality in 9.12.2, incredible effort folks!! Thank you so much!
In exploring what it can do, I have hit upon a problem I cannot solve, so hoping for some advice as I may not have the understanding I need. 

When you setup a new flow, specifically the Event Hook, and specify a scope that includes more than one event type, e.g. create, update, delete, how can you determine what event caused the trigger? 

I ask, as the payload is quite different for each event and flow logic would assist in varied processing determined by event type. 

My initial thought was, perhaps I just have to create flows for each event type, but that would result in many more flows. Whereas the ability to have one flow that can cater for multiple CRUD events would be far more elegant. 
A thought was - it would be super helpful for the $trigger object to contain an attribute that specifies the event that caused the trigger?
E.g.  
```
{
  "payload": {
    "status": "published",
    "name": "A New Thing",
    "description": "Info about the new thing"
},
  "collection": "some_things",
  "event": "items.create"
}

```

**Comment by benhaynes** [comment]
Excellent feedback! I agree that this metadata payload is very useful, maybe even required to properly define logic for those triggers. I'll wait for feedback from the engineers, but wanted to thank you for opening the discussion.

**Comment by rijkvanzanten** [comment]
Tracking the implementation of this in #14165 🙂
